### PR TITLE
Add core.isDebug() to check whether actions_step_debug is on or not.

### PR DESCRIPTION
**Pull request by TingluoHuang**
https://github.com/actions/runner/pull/253

https://github.com/actions/toolkit/pull/206

### CODE REVIEW

File: docs/action-debugging.md [hunk 1]
```diff
@@ -1,9 +1,23 @@
 # Debugging
-If the build logs do not provide enough detail on why a build may be failing, some other options exist to assist with troubleshooting.
+If the job logs do not provide enough detail on why a job may be failing, some other options exist to assist with troubleshooting.
+
+## Step Debug Logs
+This is the primary way for customer to debug their job failure that caused by certain step failed.
```
Comment (aibaars) [inline docs/action-debugging.md]: I'd suggest changing to:
This is the primary way for customers to debug job failures caused by step failures.

File: docs/action-debugging.md [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+
+Step debug logs increase the verbosity of a job's logs during and after a job's execution to assist with troubleshooting.
+
+Additional log events with the prefix `::debug::` will now also appear in the job's logs, these log events are provided by action's author and the runner process.
```
Comment (aibaars) [inline docs/action-debugging.md]: `by action's author` --> `by the Action's author`  or `by Action authors`

File: docs/action-debugging.md [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+
+### How to Access Step Debug Logs
+This flag can be enabled by [setting the secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#creating-encrypted-secrets) `ACTIONS_STEP_DEBUG` to `true`.
+
+All actions ran while this secret is enabled will show debug events in the [Downloaded Logs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/managing-a-workflow-run#downloading-logs-and-artifacts) and [Web Logs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/managing-a-workflow-run#viewing-logs-to-diagnose-failures).
 
 ## Runner Diagnostic Logs
 Runner Diagnostic Logs provide additional log files detailing how the Runner is executing an action.
 
+You need the runner diagnostic logs only if you think there is an infrastructure problem with GitHub actions and you want the product team to check the logs.
```
Comment (aibaars) [inline docs/action-debugging.md]: Suggest to change:
`GitHub actions` --> `GitHub Actions`
I think I've seen this capitalisation in the docs.

File: docs/action-debugging.md [hunk 4]
```diff
@@ -1,9 +1,23 @@
 # Debugging
-If the build logs do not provide enough detail on why a build may be failing, some other options exist to assist with troubleshooting.
+If the job logs do not provide enough detail on why a job may be failing, some other options exist to assist with troubleshooting.
+
+## Step Debug Logs
+This is the primary way for customers to debug job failures caused by step failures.
```
Comment (aibaars) [inline docs/action-debugging.md]: Apologies, now I'm re-reading my own suggestion, I find that the double use of `failures` read a bit awkwardly.  Replacing `step failures` with `failed steps` makes the sentence sound better to me. .

**Comment by bryanmacfarlane** [comment]
LGTM

File: packages/core/src/core.ts [hunk 5]
```diff
@@ -102,6 +102,13 @@ export function setFailed(message: string): void {
 // Logging Commands
 //-----------------------------------------------------------------------
 
+/**
+ * Gets whether Actions Step Debug is on or not
+ */
+export function isDebug(): boolean {
+  return process.env['RUNNER_DEBUG'] === '1'
```
Comment (aibaars) [inline packages/core/src/core.ts]: @TingluoHuang I have set the `ACTIONS_RUNNER_DEBUG` secret, but the environment variable `RUNNER_DEBUG` does not seem to be set. Am I doing something wrong, or is this pending some update of the actions runner?

Comment (bryanmacfarlane) [reply packages/core/src/core.ts]: It's in a runner that just started rolling out.

Comment (aibaars) [reply packages/core/src/core.ts]: @TingluoHuang looks like this may be ready to merge now. https://github.com/actions/runner/releases/tag/v2.165.2

Comment (TingluoHuang) [reply packages/core/src/core.ts]: yes.

**Comment by TingluoHuang** [comment]
@bryanmacfarlane this PR should be good to merged as the runner is released to all Rings.

**Comment by aibaars** [comment]
@bryanmacfarlane @TingluoHuang Can this be merged please?

**Comment by bryanmacfarlane** [comment]
I'll try and do a release today

**Comment by bryanmacfarlane** [comment]
released as core version 1.2.3

**Pull request merged**